CMSgov/mdct-mcr
# refactor form & form fields; add relevant tests

## Description

I had a bit of a lightbulb moment and realized we could leverage some features of react-hook-form a bit better than we were. It doesn't make a huge difference for the current implementation but will make a bigger difference when we start making larger, more complex nested forms.

Also had to add a bunch of tests. 👍 

### How to test

1. Pull and run.
2. Log in as admin.
3. Nav to /admin and play with the admin banner form (incl. replace, delete, etc.). Everything should work as before, but the validation and state management is improved.

### Changed Dependencies

Upgraded a testing-library util (user-event) to v14.

## Code author checklist

## Reviewer checklist (two different people)
